oxidation/crates/parsec_client_types/src/local_manifest.rs

```diff
@@ -0,0 +1,347 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use chrono::{DateTime, Utc};
+use parsec_api_crypto::SecretKey;
+use parsec_api_types::*;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use serde_with::serde_as;
+use std::collections::{HashMap, HashSet};
+use std::num::NonZeroU64;
+
+use crate::maybe_field;
+
+pub trait LoadDump
+where
+    Self: Sized + Serialize + DeserializeOwned,
+{
+    fn dump_and_encrypt(&self, key: &SecretKey) -> Vec<u8> {
+        let serialized = rmp_serde::to_vec_named(&self).unwrap_or_else(|_| unreachable!());
+        key.encrypt(&serialized)
+    }
+    fn decrypt_and_load(encrypted: &[u8], key: &SecretKey) -> Result<Self, &'static str> {
+        let serialized = key.decrypt(encrypted).map_err(|_| "Invalid encryption")?;
+        rmp_serde::from_read_ref::<_, Self>(&serialized).map_err(|_| "Invalid serialization")
+    }
+}
+
+/*
+ * Chunk
+ */
+
+// Represents a chunk of a data in file manifest.
```

Comment:
those comments should be within the struct given they describe the different attributes